hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -305,7 +305,7 @@ HoodieCompactionPlan generateCompactionPlan(
           return new CompactionOperation(dataFile, partitionPath, logFiles,
               config.getCompactionStrategy().captureMetrics(config, s));
         })
-        .filter(c -> !c.getDeltaFileNames().isEmpty()), partitionPaths.size()).stream()
+        .filter(c -> !c.getDeltaFileNames().isEmpty()), 1).stream()
```

Comment:
Is this right ? The `HoodieFlinkEngineContext#flatMap` and `HoodieJavaEngineContext#flatMap` ignores the parallelism totally.